src/Microsoft.ML.Dnn/DnnCatalog.cs

```diff
@@ -113,6 +116,9 @@ public static class DnnCatalog
             int epoch = 100,
             int batchSize = 10,
             float learningRate = 0.01f,
+            bool enableEarlyStopping = true,
```

Comment:
Do we really need a `enableEarlyStopping` parameter? What if we instead used nullable `earlyStoppingminDelta` and `earlyStoppingPatience` parameters, whose default value is `null`. If the user doesn't supply those values, then early stopping isn't enabled.

Reply:
According to @codemzs , we do not want the user to have to set any of these parameters, and want the default values to work well in most of the cases. If the default is set to null, the users would have to appropirately set these values to make use of this feature.

Reply:
According to @Zeeshan Siddiqui , we do not want the user to have to set any of these parameters, and want the default values to work well in most of the cases. If the default is set to null, the users would have to appropirately set these values to make use of this feature.  

---
In reply to: [327289504](https://github.com/dotnet/machinelearning/pull/4237#discussion_r327289504) [](ancestors = 327289504)

Reply:
I'm not convinced this is the best API to enable this. What happens if we want to enable a different stopping criteria in the future?

It feels like we should consider a different API to enable this. Check out the EarlyStoppingCriteria in FastTree. That seems like more of an extensible/future-proof API.

https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.FastTree/Training/EarlyStoppingCriteria.cs#L34-L38

Reply:
@eerhardt @ashbhandare We spoke about this on Friday and I suggested we have a class object that defines early stopping criteria, this class should extend an interface that defines bool ShouldStop(...). The API parameter should be a reference to this interface EarlyStoping and if it is set to null then we don't apply early stopping but by default it can be set to XYXEarlyStoping ...